replica/table.cc

```diff
@@ -1216,6 +1217,21 @@ table::make_memtable_list() {
     return make_lw_shared<memtable_list>(std::move(seal), std::move(get_schema), _config.dirty_memory_manager, _stats, _config.memory_compaction_scheduling_group);
 }
 
+compaction_group::compaction_group(table& t)
+    : _t(t)
+    , _table_state(std::make_unique<table_state>(t))
```

Comment:
Where is compaction_group::table_state defined?

Reply:
Justed moved table::table_state to compaction group, so it's still defined in replica/table.cc while being renamed. One of the reasons is that it makes sense, and another reason is to avoid including database.hh as I cannot forward declare a nested class type.